sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -830,11 +838,60 @@ public void populateDisplayData(DisplayData.Builder builder) {
       inner.populateDisplayData(builder);
     }
 
+    private boolean hasStatementAndSetter() {
+      return inner.getStatement() != null && inner.getPreparedStatementSetter() != null;
+    }
+
     @Override
     public PDone expand(PCollection<T> input) {
+
+      if (input.hasSchema() && !hasStatementAndSetter()) {
+        Schema schema = input.getSchema();
+
+        inner = inner.withStatement(generateStatement(inner.getTable(), schema.getFields()));
+        inner = inner.withPreparedStatementSetter(((element, preparedStatement) -> {
+          // assuming we have PCollection of <Row>. We don't have Row then we will use toRowFunction().
+          // TODO: use toRowFunction() to convert other data types into Row
+          mapAccordingToSchema((Row) element, preparedStatement, schema); // Type casting element into Row is temporary untill we implement usage of toRowFunction()
+        }));
+      }
+
       inner.expand(input);
       return PDone.in(input.getPipeline());
     }
+
+    private void mapAccordingToSchema(Row element, PreparedStatement preparedStatement, Schema schema) {
+      List<Schema.Field> fields = schema.getFields();
+      IntStream.range(0, fields.size()).forEach((index) -> {
+        JdbcUtil.setFieldPreparedStatement(element, preparedStatement, fields.get(index).getType(), index);
+      });
+    }
+
+    private String generateStatement(String tableName, List<Schema.Field> fields) {
+      final StringBuilder statement = new StringBuilder();
+      statement.append("INSERT INTO ")
+              .append(tableName)
+              .append(" (");
+
+      IntStream.range(0, fields.size()).forEach((index) -> {
+        statement.append(fields.get(index).getName());
+        if (index < fields.size() -1) {
+          statement.append(", ");
+        }
+      });
+
+      statement.append(") VALUES (");
+
+      IntStream.range(0, fields.size()).forEach((index) -> {
+        if (index == fields.size() -1)
+          statement.append("?");
+        else
+          statement.append("?, ");
+      });
+      statement.append(")");
+      return statement.toString();
+    }
+
```

Comment:
the method `generateStatement()` returns a statement for example:
INSERT INTO BEAMTEST_UT_WRITE_2019_06_11_19_31_41_569 (id, name) VALUES (?, ?)

The id and name in the above example are schema field names and we assume that schema field names will be same as database column names

Reply:
You should check that's the case by calling `#assignableTo` on the schemas